db/db_impl/db_impl_open.cc

```diff
@@ -59,6 +60,9 @@ DBOptions SanitizeOptions(const std::string& dbname, const DBOptions& src,
     if (!s.ok()) {
       // No place suitable for logging
       result.info_log = nullptr;
+      if (logger_creation_s) {
+        *logger_creation_s = s;
```

Comment:
We have ASSERT_STATUS_CHECKED failures; if the simplest solution is to PermitUncheckedError() here, that's OK with me (not ideal)

Reply:
A single `PermitUncheckedError()` here does not fully fix the issue. We can either:
- Put a PermitUncheckedError() in ~DBImpl(), or
- For each of CompactedDB, DBImplSecondary, DBImplReadOnly, explicitly call `PermitUncheckedError()` shortly after we are done with logger creation.

The latest commit goes by the second approach.

Reply:
To keep this fix simple, updated the PR to take the first approach.